akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -218,8 +218,8 @@ final class Flow[-In, +Out, +Mat](
    * It is recommended to use the internally optimized `Keep.left` and `Keep.right` combiners
    * where appropriate instead of manually writing functions that pass through one of the values.
    */
-  def joinMat[I2, O2, Mat2, M](bidi: Graph[BidiShape[Out, O2, I2, In], Mat2])(
-      combine: (Mat, Mat2) => M): Flow[I2, O2, M] = {
+  def joinMat[I2, O1, Mat2, M](bidi: Graph[BidiShape[Out, O1, I2, In], Mat2])(
+      combine: (Mat, Mat2) => M): Flow[I2, O1, M] = {
```

Comment:
Good, this makes sense. The actual out in the `FlowShape` returned is `newBidiShape.out1`

Reply:
seems correct given that BidiShape is illustrated as:
```
 *        +------+
 *  In1 ~>|      |~> Out1
 *        | bidi |
 * Out2 <~|      |<~ In2
 *        +------+
```

@heraklos Sorry for delay in reviewing this. There is the same problem in javadsl.Flow. Could you fix that at the same time and then we can merge.

Reply:
Thank you for pointing out!
Fixed :)

Reply:
Thank you for pointing out!
Fixed :)